chain/client/src/client.rs

```diff
@@ -1043,32 +1034,26 @@ impl ClientActor {
         let validators =
             self.runtime_adapter.get_epoch_block_producers(&head.epoch_id, &prev_hash)?;
         let total_validators = validators.len();
-        let prev_same_bp = self.runtime_adapter.get_block_producer(&head.epoch_id, last_height)?
+        let prev_same_bp = self.runtime_adapter.get_block_producer(&head.epoch_id, head.height)?
             == block_producer.account_id.clone();
         // If epoch changed, and before there was 2 validators and now there is 1 - prev_same_bp is false, but total validators right now is 1.
         let total_approvals =
             total_validators - min(if prev_same_bp { 1 } else { 2 }, total_validators);
-        let elapsed = self.last_block_processed.elapsed();
-        if self.approvals.len() < total_approvals
-            && elapsed < self.config.max_block_production_delay
+        if head.height > 0
+            && self.approvals.len() < total_approvals
+            && elapsed_since_last_block < self.config.max_block_production_delay
```

Comment:
Would be cleaner to pass a boolean whether to wait for more approvals into `produce_block` instead of the elapsed time (i.e. move this computation into the caller).

Reply:
why would this be cleaner? The code above doesn't concern on the rules of block production or expiration of timing to skip the approvals or not.